### PR TITLE
Hdrp/staging pt5

### PR DESCRIPTION
**Pull request by eturquin**
Path tracing improvements:
- support for sphere lights (point lights with a radius)
- support of the cast_shadow flag for visibility computations
- improved (although still limited) handling of scene dirtiness, to reset iterations
- added transmittance color support on Lit surfaces
- fixed doc (no reference to tier 3 anymore)

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl [hunk 1]
```diff
@@ -93,15 +93,15 @@ void ClosestHit(inout RayIntersection rayIntersection : SV_RayPayload, Attribute
...
+    inputSample.x = GetSample(rayIntersection.pixelCoord, _RaytracingFrameIndex, 4 * currentDepth);
+    inputSample.y = GetSample(rayIntersection.pixelCoord, _RaytracingFrameIndex, 4 * currentDepth + 1);
+    inputSample.z = GetSample(rayIntersection.pixelCoord, _RaytracingFrameIndex, 4 * currentDepth + 2);
 
     // Get current path throughput
     float3 pathThroughput = rayIntersection.color;
 
     // And reset the ray intersection color, which will store our final result
-    rayIntersection.color = computeDirect ? builtinData.emissiveColor * GetCurrentExposureMultiplier() : 0.0;
+    rayIntersection.color = computeDirect ? builtinData.emissiveColor : 0.0;
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl]: Be sure that you check this @remi-chapelain in your test (i.e that emissive color have the correct value)

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl [hunk 2]
```diff
@@ -194,52 +194,32 @@ void ClosestHit(inout RayIntersection rayIntersection : SV_RayPayload, Attribute
...
     }
 
 #else // HAS_LIGHTLOOP
-    rayIntersection.color = (!currentDepth || computeDirect) ? builtinData.emissiveColor * GetCurrentExposureMultiplier() : 0.0;
+    rayIntersection.color = (!currentDepth || computeDirect) ? builtinData.emissiveColor : 0.0;
 #endif
 
     // Bias the result (making it too dark), but reduces fireflies a lot
-    float intensity = Luminance(rayIntersection.color);
+    float intensity = Luminance(rayIntersection.color) * GetCurrentExposureMultiplier();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl]: Be sure that we have the correct emissive / light color @remi-chapelain . Maybe ask also @laurenth-unity to validate?

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl [hunk 3]
```diff
@@ -194,52 +194,32 @@ void ClosestHit(inout RayIntersection rayIntersection : SV_RayPayload, Attribute
...
-    bool isVisible = GetSurfaceDataFromIntersection(fragInput, -WorldRayDirection(), posInput, currentVertex, rayIntersection.cone, surfaceData, builtinData);
-
-    // If this fella should be culled, then we cull it
-    if (!isVisible)
-        IgnoreHit();
-
-    // If the depth information is marked as invalid, we are shooting a transmission ray
-    if (rayIntersection.remainingDepth > _RaytracingMaxRecursion)
-        AcceptHitAndEndSearch();
+    AcceptHitAndEndSearch();
```
Comment (sebastienlagarde) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl]: is it expected? we don't handle alpha clip anymore if we do this right?

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl]: Alpha clip won't work anymore with this yes

Comment (eturquin) [reply com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/ShaderPass/ShaderPassPathTracing.hlsl]: Correct, although it already wasn't supported, as we were bypassing AnyHit() altogether until now. The simplified version is here to be used when shooting shadow rays, so as to support the cast_shadows flag. If we want to handle alpha clipping, we will have to change that and use AnyHit all the time, in a future PR.

**Pull request closed without merging**